### PR TITLE
Rename GuildMemberNickChangeEvent to GuildMemberUpdateNicknameEvent

### PR DESCRIPTION
**Pull request by averen**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This PR renames `GuildMemberNickChangeEvent` to `GuildMemberUpdateNicknameEvent` and some related methods. I [discussed this change](https://discordapp.com/channels/125227483518861312/541288051243089920/543471466361126922) in the JDA discord about a two months ago and forgot to create a PR.

### CODE REVIEW

**Comment by MinnDevelopment** [comment]
The naming convention is `EntityUpdatePropertyEvent`

File: src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java [hunk 1]
```diff
@@ -24,14 +24,14 @@
  *
  * <p>Can be used to retrieve members who change their nickname, triggering guild, the old nick and the new nick.
  */
-public class GuildMemberNickChangeEvent extends GenericGuildMemberEvent
+public class GuildMemberUpdateNicknameEvent extends GenericGuildMemberEvent
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java]: This should implement `UpdateEvent` as well then

Comment (averen) [reply src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java]: Should `getPropertyIdentifier()` return `nick` or `nickname`?

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java]: Following the api docs, it should be `nick`

File: src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java [hunk 2]
```diff
@@ -40,18 +41,48 @@ public GuildMemberNickChangeEvent(JDA api, long responseNumber, Member member, S
...
-    public String getNewNick()
+    public String getNewNickname()
+    {
+        return getNewValue();
+    }
+
+    @Override
+    public String getPropertyIdentifier()
+    {
+        return "nick";
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/events/guild/member/GuildMemberUpdateNicknameEvent.java]: This should be a constant in this class, compare to other classes that implement the `UpdateEvent` interface. Additionally it should be in the class-level docs.

**Comment by DV8FromTheWorld** [comment]
@MinnDevelopment do you think that we should go ahead and add the `GenericGuildMemberUpdateEvent` hierarchy in preparation for there possibly being other member update events?

There haven't been thus far, but to do so later would be a breaking change.

**Comment by MinnDevelopment** [comment]
@DV8FromTheWorld yes that sounds reasonable, however we don't need to do that in this PR.

**Comment by DV8FromTheWorld** [comment]
Agreed. I'll put an issue into the backlog so we handle before v4 alpha is done.

**Comment by averen** [comment]
Just noticed I forgot to update the method name.

**Pull request merged**